src/components/custom/cookie-banner.tsx

```diff
@@ -115,28 +89,44 @@ const LinkText = styled(LocalizedLinkText)`
     }
 `
 
-const CookieBanner = ({ onAccept, onDecline, is_open }: CookieBannerProps) => {
-    return (
-        <Wrapper is_open={is_open}>
-            <StyledText>
-                <Localize translate_text="Cookies help us to give you a better experience and personalised content on our site. " />
-                <Localize
-                    translate_text="If you agree to our use of cookies, click on Accept. For more information, <0>see our policy</0>."
-                    components={[
-                        <LinkText key={0} to="/terms-and-conditions/#clients" color="red" />,
-                    ]}
-                />
-            </StyledText>
-            <Flex>
-                <StyledButton tertiary onClick={onDecline} mr="0.8rem">
-                    {localize("Don't accept")}
-                </StyledButton>
-                <StyledButton secondary onClick={onAccept}>
-                    {localize('Accept')}
-                </StyledButton>
-            </Flex>
-        </Wrapper>
-    )
+const CookieBanner = () => {
+    const cookie = useCookieBanner()
+    const [is_visible, setIsVisible] = React.useState(false)
+
+    React.useEffect(() => {
+        if (is_visible !== cookie.should_show) {
+            if (cookie.should_show) {
+                setIsVisible(cookie.should_show)
+            } else {
+                setTimeout(() => setIsVisible(cookie.should_show), 200)
```

Comment:
In order to show a fade out animation when the dialog is closing we have to duplicated the `should_show` state and add a delay to it, Later we can improve this.

Reply:
So `setIsVisible` here is false in both **if** statements?

Reply:
`is_visible` is just a copy of `should_show`, If `should_show` is `true` then `is_visible` is also `true` and if `should_show` is `false` then `is_visible` is also `false`, The only thing is that we are adding a delay while setting `is_visible` to false and wait for the animation to finish.